### PR TITLE
bugfix: Saga statemachine definition json cannot enable jackson parser, and when no choice matched in choice state will throw NPE

### LINKED ISSUE #3307: bug: saga statemachine definition json cannot enable jackson parser
### Ⅰ. Issue Description
Config sagaJsonParser=jackson, it is only enabled to parameter serializer json parser, but not enabled to statemachine definition json parser.

### LINKED ISSUE #3308: bug: Saga when no choice matched in choice state will throw NPE
### Ⅰ. Issue Description
Saga when no choice matched in choice state will throw NPE

### Ⅱ. Describe what happened
D8A0F042-A571-4277-BBE5-85B8146AAD41

### PR DESCRIPTION
**Pull request by long187**
### Ⅰ. Describe what this PR did
fixbug: Saga statemachine definition json cannot enable jackson parser, 
fixbug: when no choice matched in choice state will throw NPE

### Ⅱ. Does this pull request fix one issue?
fixes #3307 
fixes #3308

### CODE REVIEW

File: test/src/test/java/io/seata/saga/engine/db/StateMachineDBTests.java [hunk 1]
```diff
@@ -107,6 +107,26 @@ public void testSimpleStateMachineWithChoice() {
...
+        Map<String, Object> paramMap = new HashMap<>();
+        paramMap.put("a", 3);
+
+        String stateMachineName = "simpleChoiceNoDefaultTestStateMachine";
+
+        try {
+            stateMachineEngine.start(stateMachineName, null, paramMap);
+        } catch (EngineExecutionException e) {
+            e.printStackTrace(System.out);
+            Assertions.assertNotNull(e);
```
Comment (wangliang181230) [inline test/src/test/java/io/seata/saga/engine/db/StateMachineDBTests.java:124]: Useless code?

Comment (long187) [reply test/src/test/java/io/seata/saga/engine/db/StateMachineDBTests.java:124]: No, this unit test is expected to throw an exception

Comment (wangliang181230) [reply test/src/test/java/io/seata/saga/engine/db/StateMachineDBTests.java:124]: I mean this line of code.I mean this line of the code：`Assertions.assertNotNull(e);`

Comment (long187) [reply test/src/test/java/io/seata/saga/engine/db/StateMachineDBTests.java]: fixed

File: saga/seata-saga-statelang/src/main/java/io/seata/saga/statelang/parser/impl/StateMachineParserImpl.java [hunk 2]
```diff
@@ -42,6 +42,12 @@
 
     private static final Logger LOGGER = LoggerFactory.getLogger(StateMachineParserImpl.class);
 
+    public StateMachineParserImpl(String jsonParserName) {
```
Comment (caohdgege) [inline saga/seata-saga-statelang/src/main/java/io/seata/saga/statelang/parser/impl/StateMachineParserImpl.java:45]: 构造方法是不是应该放到字段定义后面？

Comment (long187) [reply saga/seata-saga-statelang/src/main/java/io/seata/saga/statelang/parser/impl/StateMachineParserImpl.java:47]: fixed

**Review by wangliang181230 — APPROVED** [review]
LGTM

**Review by slievrly — APPROVED** [review]
LGTM

**Pull request merged**